### PR TITLE
feat: implement EnemyAI

### PR DESCRIPTION
**Pull request by aurelenc**
Add EnemyAI component with 2 basic AIs:
- Idle AI (the entity doesn't move)
- BasicUpDown (the entity switches direction between up and down every 2 seconds)

Add RunAI system which runs the AIs

### CODE REVIEW

File: src/ecs/components/EntityAI.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+         * @param type the AIType of the entity
+         */
+        EntityAI(const AIType &type = Idle) : _thisAI(findAI(type)) {}
+
+        /**
+         * @brief get the next direction of the entity and cycles through the directions vector
+         *
+         * @return the next direction of the entity as a std::pair of chars
+         */
+        const std::pair<char, char> &getNextDirection()
```
Comment (Cavonstavant) [inline src/ecs/components/EntityAI.hpp:38]:
```suggestion
        inline const std::pair<char, char> &getNextDirection()
```

File: src/ecs/components/EntityAI.hpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            /**
+             * @brief AI copy constructor
+             *
+             * @param other another AI
+             */
+            AI(const AI &other) : currentIndex(other.currentIndex), delay(other.delay), direction(other.direction) {}
+
+            std::vector<std::pair<char, char>> direction;
+            int currentIndex;
+            short delay;
```
Comment (Cavonstavant) [inline src/ecs/components/EntityAI.hpp:81]: private members should be preceeded by an underscore
```suggestion
            std::vector<std::pair<char, char>> _direction;
            int _currentIndex;
            short _delay;
```

File: src/ecs/components/EntityAI.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#include "EntityAI.hpp"
+
```
[the remark this answers is not in the document]
Comment (aurelenc) [reply src/ecs/components/EntityAI.cpp:2]: :rage:

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
minor changes needed

File: src/ecs/components/EntityAI.hpp (hunk 1 above)
Comment (aurelenc) [reply src/ecs/components/EntityAI.hpp:38]: this function is not one line long

File: src/ecs/components/EntityAI.hpp (hunk 2 above)
Comment (aurelenc) [reply src/ecs/components/EntityAI.hpp:81]: these are not private elements
they are public elements from a struct which itself is a private member of it's parent

**Review by Bootoyka — APPROVED** [review]
Lgtm

**Pull request merged**